runtimes/runtimes/shared-registry.ts

```diff
@@ -0,0 +1,33 @@
+import { SharedRegistry } from '../server-interface/shared-registry'
+
+type SharedHandler<T, R> = {
+    name: string
+    invoke: (input: T) => Promise<R>
+}
+
+export const newSharedRegistry = (): SharedRegistry => {
+    const sharedHandlers: Record<string, SharedHandler<any, any>> = {}
+
+    return {
+        registerHandler: <T, R>(name: string, handler: (input: T) => Promise<R>) => {
```

Comment:
How is this different than an `EventEmitter`? Should we use that interface instead of creating a bespoke one?